src/SPI.cpp

```diff
@@ -142,19 +146,75 @@ void Reset()
     fclose(f);
 
     // take a backup
-    char firmbkp[1028];
+    char fwBackupPath[sizeof(FirmwarePath) + 4];
     int fplen = strlen(FirmwarePath);
-    strncpy(&firmbkp[0], FirmwarePath, fplen);
-    strncpy(&firmbkp[fplen], ".bak", 1028-fplen);
-    firmbkp[fplen+4] = '\0';
-    f = Platform::OpenLocalFile(firmbkp, "rb");
-    if (f) fclose(f);
+    strcpy(&fwBackupPath[0], FirmwarePath);
+    strncpy(&fwBackupPath[fplen], ".bak", sizeof(fwBackupPath) - fplen);
+    fwBackupPath[fplen+4] = '\0';
+    f = Platform::OpenLocalFile(fwBackupPath, "rb");
+    if (!f)
+    {
+        f = Platform::OpenLocalFile(fwBackupPath, "wb");
+        if (f)
+        {
+            fwrite(Firmware, 1, FirmwareLength, f);
+            fclose(f);
+        }
+        else
+        {
+            printf("Could not write firmware backup!\n");
+        }
+    }
     else
     {
-        f = Platform::OpenLocalFile(firmbkp, "wb");
-        fwrite(Firmware, 1, FirmwareLength, f);
         fclose(f);
     }
+}
+
+void LoadUserSettingsFromConfig() {
```

Comment:
`{` on newline please.

```suggestion
void LoadUserSettingsFromConfig()
{
```